src/contexts/productContext.js

```diff
@@ -0,0 +1,36 @@
+import { createContext, useReducer, useContext } from "react"
+
+import { productReducerFn } from "../utils/productReducerFn";
+
+const ProductContext = createContext();
+
+const ProductProvider = ({ children }) => {
+    const [state, dispatch] = useReducer(productReducerFn, {
+        sortBy: "",
+        rating: {
+            four: false,
+            three: false,
+            two: false,
+            one: false
+        },
+        category: {
+            monitor: false,
+            headphone: false,
+            keyboard: false,
+        },
```

Comment:
This could be optimized to only one no need to specify differently for different products